spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapterTests.java

```diff
@@ -31,13 +47,166 @@
 @RunWith(MockitoJUnitRunner.class)
 public class PubSubInboundChannelAdapterTests {
 
-	@Mock
-	private PubSubTemplate pubSubTemplate;
+	public static final String ACK = "ACK";
+
+	public static final String NACK = "NACK";
 
 	@Test(expected = IllegalArgumentException.class)
 	public void testNonNullAckMode() {
+		PubSubTemplate pubSubTemplate = Mockito.mock(PubSubTemplate.class);
+
 		PubSubInboundChannelAdapter adapter = new PubSubInboundChannelAdapter(
-				this.pubSubTemplate, "testSubscription");
+				pubSubTemplate, "testSubscription");
+
 		adapter.setAckMode(null);
 	}
+
+	@Test
+	public void testAckModeAuto() throws Exception {
+		PubSubTemplate pubSubTemplate = Mockito.mock(PubSubTemplate.class);
+		QueueChannel queueChannel = Mockito.mock(QueueChannel.class);
+
+		final Holder<String> returnValue = new Holder<>();
```

Comment:
What is this `Holder`, please?
Why standard Java `AtomicReference` is not enough?